packages/ember-intl/addon/services/intl.ts

```diff
@@ -96,7 +101,15 @@ export default class IntlService extends Service {
     // eslint-disable-next-line prefer-rest-params
     super(...arguments);
 
-    this.getDefaultFormats();
+    const hasNewConfiguration = Boolean(
+      // @ts-expect-error: Property 'resolveRegistration' does not exist on type 'Owner'
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      getOwner(this).resolveRegistration('ember-intl:main'),
```

Comment:
I would let users decide where to store their formats, and not force them to use a specific file. Otherwise, I don't see how this is any different from the deprecated `app/formats.js` file. This solution also uses the resolver, which is something the Ember team wants to get rid of if I'm not mistaken.

Reply:
```
Currently, the path app/ember-intl.{js,ts} does matter, because the check for the new configuration file uses getOwner(this).resolveRegistration('ember-intl:main'). ember-intl may allow arbitrary paths (e.g. app/configs/ember-intl.{js,ts}) in v8 (TBD), since people can explicitly import their formats, then pass it to ember-intl via setFormats().
```

Missed this in the description. Migration path still feels a bit weird though.

Reply:
No problem. Yeah, a possible extra step in migration isn't ideal; I didn't think of an easier way to check for the existence of the new configuration file, which would allow people to start creating this file in `>=7.3.0`.